utils/land-patterns.stanza

```diff
@@ -18,6 +20,58 @@ defn ensure-even-positive! (value: Int, name: String):
   if value < 2 or value % 2 == 1:
     fatal("%_ only supports a positive even number of pins." % [name])
 
+; ====== Silk screen helpers ===================================================
+; Create a polarity marker relative to a pad.
+; 
+; pad-1: the pad to mark for the polarity 
+; part-dims?: optional. If provided, the generator will use these to determine where
+;             to place the pad, and guarantee it is not underneath the part.
+public defn draw-polarity-marker (pad-1:JITXObject, part-dims?:Dims|False) :
+  val min-silk-width = clearance(current-rules(), MinSilkscreenWidth)
+  inside pcb-landpattern:
+    ; draw a line of `min-silk-width` width and length 3.0 * min-silk-width 
+    ; centered at origin
+    defn draw-line (origin:Pose):
+      val line-width  = min-silk-width
+      val line-length = line-width * 3.0
+      val line = Line(line-width, [
+        Point(min-silk-width * 0.5, 0.0)
+        Point(line-width + min-silk-width * 0.5, 0.0)  
+      ])
+      layer(Silkscreen("pol", Top)) = origin * line
+
+    defn part-pose () :
```

Comment:
I'm not sure what the `part-dims?` argument is controlling here, can you describe it?